File: typescript/src/models/AssetEventGroup.ts
/**
 * AssetEventGroup model
 * Converted from go/models/asset_event_group.go
 * Owner: backend (event grouping)
 * Consumers: frontend, event-correlation-service
 */

/** GeoJSONPoint represents a geographical point in GeoJSON format */
export interface GeoJSONPoint {
  type: string; // Always "Point"
  coordinates: [number, number]; // [longitude, latitude]
}

/** Event represents a single event within a group */
export interface Event {
  id: string;
  type: string;
  timestamp: string; // ISO 8601
  location?: GeoJSONPoint;
  severity: string;
  description: string;
  metadata?: Record<string, any>;
}

/** AssetEventGroup represents grouped events by asset (camera or fire_detector) */
export interface AssetEventGroup {
  assetId: string;
  assetName: string;
  assetType: string; // "camera" or "fire_detector"
  eventCount: number;
  latestEvent: Event;
  eventIds: string[];
  events?: Event[]; // Full event data for display
}

/** GroupedEventsResponse is the response structure for grouped events API */
export interface GroupedEventsResponse {
  groups: AssetEventGroup[];
  count: number;
}

File: python/src/phlx_contracts/models/asset_event_group.py
"""Asset event group model for Phylax platform"""

from typing import Optional, List, Dict, Any, Tuple
from pydantic import BaseModel, Field

# Owner: backend (event grouping)
# Consumers: frontend, event-correlation-service


class GeoJSONPoint(BaseModel):
    """GeoJSON point representation"""

    type: str = "Point"
    coordinates: Tuple[float, float]  # [longitude, latitude]

    class Config:
        populate_by_name = True


class Event(BaseModel):
    """Single event within a group"""

    id: str
    type: str
    timestamp: str
    location: Optional[GeoJSONPoint] = None
    severity: str
    description: str
    metadata: Optional[Dict[str, Any]] = None

    class Config:
        populate_by_name = True


class AssetEventGroup(BaseModel):
    """Grouped events by asset"""

    asset_id: str = Field(alias="assetId")
    asset_name: str = Field(alias="assetName")
    asset_type: str = Field(alias="assetType")  # "camera" or "fire_detector"
    event_count: int = Field(alias="eventCount")
    latest_event: Event = Field(alias="latestEvent")
    event_ids: List[str] = Field(alias="eventIds")
    events: Optional[List[Event]] = None  # Full event data for display

    class Config:
        populate_by_name = True


class GroupedEventsResponse(BaseModel):
    """Response for grouped events API"""

    groups: List[AssetEventGroup]
    count: int

    class Config:
        populate_by_name = True
